mods/d2k/rules/vehicles.yaml

```diff
@@ -328,8 +328,22 @@ devastator:
 	Explodes:
 		Weapon: UnitExplodeLarge
 		EmptyWeapon: UnitExplodeLarge
+		RequiresCondition: !overload
 	SpawnActorOnDeath:
 		Actor: devastator.husk
+	Explodes@OVERLOAD:
+		Weapon: PlasmaExplosion
+		EmptyWeapon: PlasmaExplosion
+		RequiresCondition: overload
```

Comment:
This has the same issue that was raised in #13878 with regards to Demo Trucks and MAD Tanks, where it will fire this weapon regardless of whether it self-destructed or was killed normally during the triggered phase.   The fix suggested in one of the outdated comment chains there will solve it here too: have `KillsSelf` grant a condition immediately before killing the actor which can be used to trigger suicide-specific actions.